### PR TITLE
[Issue#319] Fix the nil pointer exception for missing config string

### LINKED ISSUE #319: edgecontroller panics when there is no configuration for cloudhub.ca, cloudhub.cert, cloudhub.key.
**What happened**:
edgecontroller crashes with missing configuration for cloudhub.ca, cloudhub.cert, cloudhub.key.

**What you expected to happen**:
Exit gracefully with missing configuration information.

**How to reproduce it (as minimally and precisely as possible)**:
Remove configuration for
  ca: /tmp/rootCA.crt
  cert: /tmp/edge.crt
  key: /tmp/edge.key
in controller.yaml.

**Anything else we need to know?**:
Also, need to remove the code in cloudhub.go:Start() for redundant code.

	// init filter
	pool := x509.NewCertPool()
	ok := pool.AppendCertsFromPEM(util.HubConfig.Ca)
	if !ok {
		panic(fmt.Errorf("fail to load ca content"))
	}

**Environment**:
- KubeEdge version: 0.2
- Hardware configuration: Virtual box vm - 2 vcpus / 2GB memory
- OS (e.g. from /etc/os-release): Ubuntu 18.04.1 LTS
- Kernel (e.g. `uname -a`):Linux 4.18.0-16-generic 
- Others:

Logs==>

2019/04/04 13:30:25 WARN: empty configurtion from [CommandlineSource]
2019/04/04 13:30:25 WARN: empty configurtion from [MemorySource]
New file source added for configuration: /opt/spire/app-binaries/conf/controller.yamlNew file source added for configuration: /opt/spire/app-binaries/conf/edge.yamlNew file source added for configuration: /opt/spire/app-binaries/conf/logging.yamlNew file source added for configuration: /opt/spire/app-binaries/conf/modules.yaml/opt/spire/app-binaries/conf/logging.yamllogConfig:&{INFO edgecontroller.log [file stdout] true false  }
2019/04/04 13:30:25 RollingPolicy is empty, use default policy[size]
2019-04-04 13:30:25.959 +05:30 INFO core/module.go:41 module cloudhub registered
2019-04-04 13:30:25.981 +05:30 INFO config/buffer.go:27 update pod status buffer: 1024
2019-04-04 13:30:25.981 +05:30 INFO config/buffer.go:34 Update node status buffer: 1024
2019-04-04 13:30:25.981 +05:30 INFO config/buffer.go:41 query config map buffer: 1024
2019-04-04 13:30:25.981 +05:30 INFO config/buffer.go:48 query secret buffer: 1024
2019-04-04 13:30:25.981 +05:30 INFO config/context.go:28  send module name: cloudhub
2019-04-04 13:30:25.981 +05:30 INFO config/context.go:35 receive module name: controller
2019-04-04 13:30:25.981 +05:30 INFO config/context.go:42 response module name: cloudhub
2019-04-04 13:30:25.981 +05:30 INFO config/kube.go:38 kube master: http://localhost:8080
2019-04-04 13:30:25.981 +05:30 INFO config/kube.go:45 kube config: 
2019-04-04 13:30:25.981 +05:30 INFO config/kube.go:52 kube namespace: 
2019-04-04 13:30:25.981 +05:30 INFO config/kube.go:59 kube content type: application/vnd.kubernetes.protobuf
2019-04-04 13:30:25.981 +05:30 INFO config/kube.go:66 kube QPS: 5.000000
2019-04-04 13:30:25.981 +05:30 INFO config/kube.go:73 kube burst: 10
2019-04-04 13:30:25.982 +05:30 INFO config/kube.go:80 kube update frequency: 10s
2019-04-04 13:30:25.982 +05:30 INFO config/load.go:27 update pod status workers: 1
2019-04-04 13:30:25.982 +05:30 INFO config/load.go:34 update node status workers: 1
2019-04-04 13:30:25.982 +05:30 INFO config/load.go:41 query config map workers: 4
2019-04-04 13:30:25.982 +05:30 INFO config/load.go:48 query secret workers: 4
2019-04-04 13:30:25.982 +05:30 INFO config/messagelayer.go:18 message layer: context
2019-04-04 13:30:26.020 +05:30 INFO core/module.go:41 module controller registered
2019-04-04 13:30:26.021 +05:30 INFO core/core.go:23 starting module cloudhub
panic: interface conversion: interface {} is nil, not string

goroutine 9 [running]:
github.com/kubeedge/kubeedge/cloud/edgecontroller/pkg/cloudhub.(*cloudHub).Start(0xc0000b4130, 0xc00030aca0)
	/home/root1/gopath/src/github.com/kubeedge/kubeedge/cloud/edgecontroller/pkg/cloudhub/cloudhub.go:39 +0x4fd
created by github.com/kubeedge/kubeedge/vendor/github.com/kubeedge/beehive/pkg/core.StartModules
	/home/root1/gopath/src/github.com/kubeedge/kubeedge/vendor/github.com/kubeedge/beehive/pkg/core/core.go:22 +0x153

### PR DESCRIPTION
**Pull request by trilokgm**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
The PR fixes nil pointer exception when there is no config found for cloudhub certificates and key file path. Initializes with default path values.

**Which issue(s) this PR fixes**:

Fixes #319 

**Special notes for your reviewer**:
Fix for graceful exit is now implemented with os.Exit(1) so that testcases can be enhanced for configuration validations.
A more robust framework needs to reworked w.r.t to beehive implementation and module implementations require to be adapted for error returns.

### CODE REVIEW

**Comment by rohitsardesai83** [comment]
@trilokgm , can you please resolve the conflicts ?

**Comment by rohitsardesai83** [comment]
@trilokgm , the beehive core has a graceful shutdown method which does modules cleanup. When we call os.Exit(1), this would trigger the graceful shutdown right ? is this happening now ?

**Comment by trilokgm** [comment]
> @trilokgm , the beehive core has a graceful shutdown method which does modules cleanup. When we call os.Exit(1), this would trigger the graceful shutdown right ? is this happening now ?

Beehive captures os signals and makes call which end up in context cleanup. The goroutines are not stopped in graceful shutdown call.

os.Exit(1) does not send any signal to invoke graceful shutdown execution flow. The program exits with an error code, here it is 1. There is no panic or stack trace generated.

I had given a thought about using uniform error codes, had to drop the idea due to significant usage of go routines without error notification.Due to this, there is not much significance seen for using error codes for now.

File: edge/pkg/eventbus/event_bus.go [hunk 1]
```diff
@@ -99,14 +101,16 @@ func (eb *eventbus) Start(c *context.Context) {
 		}
 
 		if qos.(int) < int(packet.QOSAtMostOnce) || qos.(int) > int(packet.QOSExactlyOnce) || sessionQueueSize.(int) <= 0 {
-			panic("mqtt.qos must be one of [0,1,2] or mqtt.session-queue-size must > 0")
+			log.LOGGER.Errorf("mqtt.qos must be one of [0,1,2] or mqtt.session-queue-size must > 0")
```
Comment (CindyXing) [inline edge/pkg/eventbus/event_bus.go]: The benefit of panic is the exception callstack. Here we log error in the module start where the error is clear. So log the error and exit is fine.

**Pull request merged**